tools/distpartitioning/globalids.py

```diff
@@ -0,0 +1,279 @@
+    part_ids = node_part_ids[global_nids]
+
+    #form list of lists, each list includes global_nids whose mappings (shuffle_global_nids) needs to be retrieved.
+    #and rank will be the process which owns mappings of these global_nids
+    global_nids_ranks = []
+    for i in range(world_size):
+        if (i == rank):
+            global_nids_ranks.append(np.empty(shape=(0)))
+            continue
+
+        #not_owned_nodes = part_ids[:,0][part_ids[:,1] == i]
+        not_owned_node_ids = np.where(part_ids == i)[0] 
+        if not_owned_node_ids.shape[0] == 0: 
+            not_owned_nodes = np.empty(shape=(0))
+        else: 
+            not_owned_nodes = global_nids[not_owned_node_ids]
+        global_nids_ranks.append(not_owned_nodes)
+
+    #Retrieve Global-ids for respective node owners
+    #shuffle_global_nids, global_nids_ranks = get_shuffle_global_nids(rank, world_size, global_nids_ranks, node_data)
+    resolved_global_nids = get_shuffle_global_nids(rank, world_size, global_nids_ranks, node_data)
+
+    #Add global_nid <-> shuffle_global_nid mappings to the received data
+    for i in range(world_size):
+        if (i == rank):
+            #own_nodeids = part_ids[:,0][part_ids[:,1] == i]
+            own_node_ids = np.where(part_ids == i)[0]
+            own_global_nids = global_nids[own_node_ids]
+            common, ind1, ind2 = np.intersect1d(node_data[:,3], own_global_nids, return_indices=True)
+            my_shuffle_global_nids = node_data[ind1,0]
+            #shuffle_global_nids.extend(my_shuffle_global_nids.tolist())
+            #global_nids_ranks.extend(own_global_nids[ind2].tolist())
+            local_mappings = np.column_stack((own_global_nids, my_shuffle_global_nids))
+            resolved_global_nids = np.concatenate((resolved_global_nids, local_mappings))
+
+    #form a dictionary of mappings between orig-node-ids and global-ids
+    resolved_mappings = dict(zip(resolved_global_nids[:,0], resolved_global_nids[:,1]))
+
+    #determine global-ids for the orig-src-id and orig-dst-id
+    shuffle_global_src_id = [ resolved_mappings[ x ] for x in edge_data[:, 0] ]
+    shuffle_global_dst_id = [ resolved_mappings[ x ] for x in edge_data[:, 1] ]
+
+    return np.c_[np.asarray(shuffle_global_src_id, dtype=np.int64), np.c_[ np.asarray(shuffle_global_dst_id, dtype=np.int64), edge_data]]
+    #return np.c_[shuffle_global_src_id, np.c_[ shuffle_global_dst_id, edge_data]]
+
+
+def assign_shuffle_global_nids_nodes(rank, world_size, ntype_counts, node_data):
+    """
+    Utility function to assign shuffle global ids to nodes at a given rank
+    node_data gets converted from [ntype, global_type_nid, global_nid]
+    to [shuffle_global_nid, ntype, global_type_nid, global_nid, part_local_type_nid]
+    where shuffle_global_nid : global id of the node after data shuffle
+            ntype : node-type as read from xxx_nodes.txt
+            global_type_nid : node-type-id as read from xxx_nodes.txt
```

Comment:
these ids are not ids per type.

Reply:
Marking this as resolved as per our code review discussion !!